### DISCUSSION TITLE
NullPointerException: because "cacheObject" is null

### DISCUSSION THREAD

**Discussion by segheim** [apache/skywalking#12257] [Q&A]
Hi! Would you help to solve problem) 
I started spring boot application using apache-skywalking-java-agent-9.1.0. When i call endpoint(post, save data in postgresql) i get NPE. 
/ERROR 2024-05-27 15:59:08.733 http-nio-9080-exec-5 InstMethodsInter : class[class org.postgresql.jdbc.PgPreparedStatement] after method[executeUpdate] intercept failure 
java.lang.NullPointerException: Cannot invoke "org.apache.skywalking.apm.plugin.jdbc.define.StatementEnhanceInfos.getConnectionInfo()" because "cacheObject" is null/

**Comment by wu-sheng** [comment]
This usually means, your way of using pg driver is not supposed.

Here are tested scenarios, 

- https://github.com/apache/skywalking-java/tree/main/test/plugin/scenarios/postgresql-scenario
- https://github.com/apache/skywalking-java/tree/main/test/plugin/scenarios/postgresql-above9.4.1207-scenario

**Reply by segheim** [reply]
org.postgresql:postgresql:42.7.2   skywalking is not support, isn't it?

**Reply by wu-sheng** [reply]
It could be version, or unsupported API, or different JDBC connection URL. I can't tell that detail.

SkyWalking agent is hosting the plugins with above test scenarios. It could be more. So, the thing is enhanced by the community version by version.

**Reply by segheim** [reply]
if I change spring boot on version 3.3.0 i have not get NPE. If spring boot version 3.2.3  I have NPE. It's interesting situation
